### DISCUSSION TITLE
How to use a custom jsonDecodeCallback with BackgroundTransformer

### DISCUSSION THREAD

**Discussion by lars-hl-knowit** [cfug/dio#2340] [Q&A]
Hello, in our Flutter project, we've set the transformer for our Dio client like this:
`dio.transformer = DefaultTransformer(
      jsonDecodeCallback: (body) {
        final transformedBody = _convertLongIntsToString(body);
        return json.decode(transformedBody);
      },
    );`

After updating the Flutter Dio package, `DefaultTransformer` is now deprecated, and one is advised to use `BackgroundTransformer` instead. But `BackgroundTransformer` does not accept a custom `jsonDecodeCallback` function. We need this, because the data our app interacts with uses numbers for IDs which are so long that when the app is built for Web, and converted to java script, these IDs are misrepresented. So we convert them to strings with our custom `jsonDecodeCallback` function.

So, is there a way to achieve the same functionality using the now recommended `BackgroundTransformer` instead of the deprecated `DefaultTransformer`?

**Comment by kuhnroyal** [comment]
The `BackgroundTransformer` extends `SyncTransformer` which you can use.
